src/main/java/de/zalando/tip/zalenium/servlet/LiveNodeHtmlRenderer.java

```diff
@@ -0,0 +1,226 @@
+        builder.append("</p>");
+
+        builder.append("<iframe src='").append(vncReadOnlyUrl).append("' class='vnc' ");
+        builder.append("style='display: inline-flex; width: 100%; height: 300px; border:none; margin:0; padding:0;'>");
+        builder.append("Your browser does not support iframes. </iframe>");
+
+        builder.append("</div>");
+        return builder.toString();
+    }
+
+    // the lines of icon representing the possible slots
+    private String getLines(SlotsLines lines) {
+        StringBuilder builder = new StringBuilder();
+        for (MiniCapability cap : lines.getLinesType()) {
+            String icon = cap.getIcon();
+            String version = cap.getVersion();
+            builder.append("<p>");
+            if (version != null) {
+                builder.append("v:").append(version);
+            }
+            for (TestSlot s : lines.getLine(cap)) {
+                builder.append(getSingleSlotHtml(s, icon));
+            }
+            builder.append("</p>");
+        }
+        return builder.toString();
+    }
+
+    // icon ( or generic html if icon not available )
+    private String getSingleSlotHtml(TestSlot s, String icon) {
+        StringBuilder builder = new StringBuilder();
+        TestSession session = s.getSession();
+        if (icon != null) {
+            builder.append("<img src='").append(icon).append("' width='16' height='16'");
+        } else {
+            builder.append("<a href='#' ");
+        }
+
+        if (session != null) {
+            builder.append(" class='busy' title='").append(session.get("lastCommand")).append("' ");
+        } else {
+            builder.append(" title='").append(s.getCapabilities()).append("'");
+        }
+
+        if (icon != null) {
+            builder.append(" />\n");
+        } else {
+            builder.append(">");
+            builder.append(s.getCapabilities().get(CapabilityType.BROWSER_NAME));
+            builder.append("</a>");
+        }
+        return builder.toString();
+    }
+
+    // the tabs header.
+    private String nodeTabs() {
+        String nodeTabs = "";
+        nodeTabs = nodeTabs.concat("<div class='tabs'>");
+        nodeTabs = nodeTabs.concat("<ul>");
+        nodeTabs = nodeTabs.concat("<li class='tab' type='browsers'><a title='test slots' href='#'>Browsers</a></li>");
+        nodeTabs = nodeTabs.concat("<li class='tab' type='config'><a title='node configuration' href='#'>Configuration</a></li>");
+        nodeTabs = nodeTabs.concat("</ul>");
+        nodeTabs = nodeTabs.concat("</div>");
+        return nodeTabs;
+    }
+
+
+    /**
+     * return the platform for the proxy. It should be the same for all slots of the proxy, so checking that.
+     * @param proxy remote proxy
+     * @return Either the platform name, "Unknown", "mixed OS", or "not specified".
+     */
+    public static String getPlatform(RemoteProxy proxy) {
+        Platform res;
+        if (proxy.getTestSlots().size() == 0) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1155)